### PR TITLE
Unit strategy

### PR DESCRIPTION
**Pull request by jochem725**
I've made some small changes to the way the strategy pattern is implemented in unit.
I've added shootableunit and activatable unit, representing units that are able to shoot and are able to be activated on the player, like powerups.

### CODE REVIEW

**Comment by bryanvanwijk** [comment]
Looks good!

File: src/main/java/spaceinvaders/group_22/unit/SpaceShip.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
  * @author Dorian
  */
 @SuppressWarnings("checkstyle:magicnumber") 
-public class SpaceShip extends Unit implements MovableUnit {
+public class SpaceShip extends Unit implements MovableUnit, ShootableUnit {
```
Comment (doriandekoning) [inline src/main/java/spaceinvaders/group_22/unit/SpaceShip.java]: Might ShootingUnit not be a better name. Shootable made me think it was a bullet or some other object that "can be shot".

**Comment by doriandekoning** [comment]
Apart from that little thing. It looks great!

**Comment by jochem725** [comment]
I've refactored the code so the name is less confusing.

**Comment by egedebruin** [comment]
Looks great

**Comment by jochem725** [comment]
I'll merge it then.

**Pull request merged**